core/spec/models/spree/preferences/preferable_spec.rb

```diff
@@ -151,6 +151,16 @@ def self.allowed_admin_form_preference_types
       end
     end
 
+    context "when a default is a proc" do
+      before do
+        A.preference(:context, :string, default: -> { self.to_s })
+      end
+
+      it "calls it from the instance" do
+        expect(@a.preferred_context).to eq(@a.to_s)
```

Comment:
I'm a bit concerned that with time this could become a false positive (pretty sure it is not now, but wanted to ask anyway): what is the content of  `@a.to_s` at this point? 

I thought that maybe we could also (or instead) test that `@a` receives `to_s` so no matter what the value is, we are sure that `self` is the instance of `A`, which is what we want to cover here. 

Let me know your thoughts.

Reply:
If I rollback the modification in `core/lib/spree/preferences/preferable_class_methods.rb` the test will fail like this:
```
expected: "#<A:0x000000010f6ff3c8>"
got: "#<RSpec::ExampleGroups::SpreePreferencesPreferable::PreferenceAccess::WhenADefaultIsAProc:0x000000010f684858>"
``` 
The `#to_s` runs in the test context, for me it looked enough

Reply:
I also thought that this test was a bit indirect. Another option that would allow us to avoid stubbing is testing for something in the state of the `A` instance. Like:

```ruby
A.preference(:context, :string, default: -> { preferred_color })
# ...
expect(@a.preferred_context).to eq('green')
```

The above would have the added benefit of covering probably the most common use case.

What do you think, @Roddoric?

Reply:
Soooo, it was a good catch as it proved the fix did not actually work...

But that lead to others problems, so I resolve this conversation and start a new one...